internal/fetch/fetch.go

```diff
@@ -156,22 +175,22 @@ func downloadAttempt(target, source, expectedSha256 string) (err error) {
 	client := http.Client{Timeout: 60 * time.Second}
 	response, err := client.Get(source)
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer response.Body.Close()
 	if response.StatusCode >= 300 {
-		return fmt.Errorf("http error in download %s", response.Status)
+		return "", fmt.Errorf("http error in download %s", response.Status)
 	}
 
 	if _, err := io.Copy(writer, response.Body); err != nil {
-		return err
+		return "", err
 	}
 
-	got := fmt.Sprintf("%x", hasher.Sum(nil))
-	if expectedSha256 != got {
-		return fmt.Errorf("mismatched hash on download, expected=%s, got=%s", expectedSha256, got)
+	sha = fmt.Sprintf("%x", hasher.Sum(nil))
+	if err = os.Rename(tempFile.Name(), target); err != nil {
+		return "", err
```

Comment:
This changes `target` before verifying the checksum.  Previously, the download was in `/tmp` until the checksum was verified.